File: ahu_dashboard/lib/models/ahu_unit.dart
/// Represents an AHU unit in the hospital
class AhuUnit {
  final String id;           // Internal unique key (e.g., "ahu-01@site/room")
  final String rawId;        // MQTT AHU topic id segment (e.g., "ahu-01")
  final String name;         // Display name
  final String site;         // Hospital site (e.g., "hospitalA")
  final String room;         // Room location (e.g., "icu1")
  final String org;          // Organization (e.g., "almed")
  
  AhuUnit({
    required this.id,
    String? rawId,
    required this.name,
    required this.site,
    required this.room,
    required this.org,
  }) : rawId = rawId ?? id;

  /// Get the MQTT base topic for this AHU
  String get baseTopic => '$org/ahu/$site/$room/$rawId';

  /// Get telemetry topic
  String get telemetryTopic => '$baseTopic/telemetry';

  /// Get state topic
  String get stateTopic => '$baseTopic/state';

  /// Get log topic
  String get logTopic => '$baseTopic/log';

  /// Get command topic
  String get cmdTopic => '$baseTopic/cmd';

  /// Get status topic
  String get statusTopic => '$baseTopic/status';

  /// Get WiFi provisioning topic
  String get provWifiTopic => '$baseTopic/provision/wifi';

  /// Get broker provisioning topic
  String get provBrokerTopic => '$baseTopic/provision/broker';

  /// Get motor timings provisioning topic
  String get provMotorTimingsTopic => '$baseTopic/provision/motor_timings';

  /// Get provisioning acknowledgment topic
  String get provAckTopic => '$baseTopic/provision/ack';

  /// Create from JSON
  factory AhuUnit.fromJson(Map<String, dynamic> json) {
    return AhuUnit(
      id: json['id'] as String,
      rawId: (json['rawId'] ?? json['id']) as String,
      name: json['name'] as String,
      site: json['site'] as String,
      room: json['room'] as String,
      org: json['org'] as String,
    );
  }

  /// Convert to JSON
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'rawId': rawId,
      'name': name,
      'site': site,
      'room': room,
      'org': org,
    };
  }
}

File: pi_dashboard/app/models/ahu_unit.py
from dataclasses import dataclass, field


@dataclass
class AhuUnit:
    id: str
    name: str
    site: str
    room: str
    org: str = "almed"

    @property
    def base_topic(self) -> str:
        return f"{self.org}/ahu/{self.site}/{self.room}/{self.id}"

    @property
    def telemetry_topic(self) -> str:
        return f"{self.base_topic}/telemetry"

    @property
    def state_topic(self) -> str:
        return f"{self.base_topic}/state"

    @property
    def log_topic(self) -> str:
        return f"{self.base_topic}/log"

    @property
    def cmd_topic(self) -> str:
        return f"{self.base_topic}/cmd"

    @property
    def status_topic(self) -> str:
        return f"{self.base_topic}/status"

    @property
    def prov_wifi_topic(self) -> str:
        return f"{self.base_topic}/provision/wifi"

    @property
    def prov_broker_topic(self) -> str:
        return f"{self.base_topic}/provision/broker"

    @property
    def prov_motor_timings_topic(self) -> str:
        return f"{self.base_topic}/provision/motor_timings"

    @property
    def prov_ack_topic(self) -> str:
        return f"{self.base_topic}/provision/ack"
